src/fsharp/CheckExpressions.fs

```diff
@@ -6936,7 +6963,7 @@ and TcRecdExpr cenv overallTy env tpenv (inherits, optOrigExpr, flds, mWholeExpr
         if isFSharpObjModelTy cenv.g overallTy then errorR(Error(FSComp.SR.tcTypeIsNotARecordTypeNeedConstructor(), mWholeExpr))
         elif not (isRecdTy cenv.g overallTy) then errorR(Error(FSComp.SR.tcTypeIsNotARecordType(), mWholeExpr))
 
-    let superTy, tpenv =
+    let superInitOpt, tpenv =
```

Comment:
nit: The name is better than `superTy` for sure as it's an expression, but `superInitExprOpt` might be clearer.

Reply:
Thanks!